bitnami/apisix/templates/data-plane/dep-ds.yaml

```diff
@@ -0,0 +1,216 @@
+{{- if .Values.dataPlane.enabled }}
+apiVersion: {{ include "common.capabilities.deployment.apiVersion" . }}
+kind: {{ ternary "DaemonSet" "Deployment" .Values.dataPlane.useDaemonSet }}
+metadata:
+  name: {{ template "apisix.data-plane.fullname" . }}
+  namespace: {{ include "common.names.namespace" . | quote }}
+  labels: {{- include "common.labels.standard" . | nindent 4 }}
+    app.kubernetes.io/part-of: apisix
+    app.kubernetes.io/component: data-plane
+    {{- if .Values.commonLabels }}
+    {{- include "common.tplvalues.render" ( dict "value" .Values.commonLabels "context" $ ) | nindent 4 }}
+    {{- end }}
+  {{- if .Values.commonAnnotations }}
+  annotations: {{- include "common.tplvalues.render" ( dict "value" .Values.commonAnnotations "context" $ ) | nindent 4 }}
+  {{- end }}
+spec:
+  {{- if and (not .Values.dataPlane.useDaemonSet) (not .Values.dataPlane.autoscaling.hpa.enabled) }}
+  replicas: {{ .Values.dataPlane.replicaCount }}
+  {{- end }}
+  {{- if .Values.dataPlane.updateStrategy }}
+  {{ ternary "updateStrategy" "strategy" .Values.dataPlane.useDaemonSet }}: {{- toYaml .Values.dataPlane.updateStrategy | nindent 4 }}
+  {{- end }}
+  selector:
+    matchLabels: {{- include "common.labels.matchLabels" . | nindent 6 }}
+      app.kubernetes.io/part-of: apisix
+      app.kubernetes.io/component: data-plane
+  template:
+    metadata:
+      annotations:
+        checksum/config: {{ include (print $.Template.BasePath "/data-plane/configmap.yaml") . | sha256sum }}
+        {{- if .Values.podAnnotations }}
+        {{- include "common.tplvalues.render" (dict "value" .Values.podAnnotations "context" $) | nindent 8 }}
+        {{- end }}
+        {{- if .Values.dataPlane.podAnnotations }}
+        {{- include "common.tplvalues.render" (dict "value" .Values.dataPlane.podAnnotations "context" $) | nindent 8 }}
+        {{- end }}
```

Comment:
Duplicated value `podAnnotations`.

If they are different values, `podAnnotations` applies everywhere while `dataPlane.podAnnotations` applies only on dataplane deployment, the `podAnnotations` is missing in the other deployment files.